### DISCUSSION TITLE
Sorting JavaScript object properties?

### DISCUSSION THREAD

**Discussion by pabloruizetc** [tailwindlabs/tailwindcss#12604] [Help] [answered]
I'm looking to sort TailwindCSS classes within JavaScript object properties, as shown here:

```js
const obj = {
  class: "hover:bg-black flex ...",
}
```

While I'm aware of sorting classes in other contexts like non-standard attributes, within function calls and in template literals, I'm unsure if it's possible with object properties. Can someone confirm if this is feasible and explain how to do it?

**Comment by dev-alihasan** [comment]
hmm it's indeed possible to sort tailwind CSS classes within JS object properties, since the classes are just strings, u ca use javascript string manipulation and array methods to achieve this.

```
const obj = {
  class: "hover:bg-black flex ...",
};

// classes from the object
const classes = obj.class.split(" ");

// sort the classes alphabetically
const sortedClasses = classes.sort();

// join the sorted classes back into a string
const sortedClassString = sortedClasses.join(" ");

// update the object with the sorted classes
obj.class = sortedClassString;

console.log(obj);

```

**Reply by pabloruizetc** [reply]
Oops, I forgot to specify that I was talking about the "Prettier plugin for Tailwind CSS that automatically sorts classes according to the recommended class order."

**Reply by wongjn** [reply, marked as answer]
You could:
- [Configure class sorting within a function](https://github.com/tailwindlabs/prettier-plugin-tailwindcss?tab=readme-ov-file#sorting-classes-in-function-calls) and then have a function that does nothing:
  ```js
  // prettier.config.js
  module.exports = {
    tailwindFunctions: ['_'],
  }
  ```
  ```js
  const _ = x => x;
  
  const obj = {
    class: _("hover:bg-black flex ..."),
  }
  ```
- [Configure class sorting within a template literal tag](https://github.com/tailwindlabs/prettier-plugin-tailwindcss?tab=readme-ov-file#sorting-classes-in-template-literals) and then have a template tag that does nothing:
  ```js
  // prettier.config.js
  module.exports = {
    tailwindFunctions: ['tw'],
  }
  ```
  ```js
  const tw = (strings, ...values) => String.raw({ raw: strings }, ...values);
  
  const obj = {
    class: tw`hover:bg-black flex ...`,
  }
  ```
- Fork [the plugin source code](https://github.com/tailwindlabs/prettier-plugin-tailwindcss) and add object string sorting.

**Reply by pabloruizetc** [reply]
Thank you so much @wongjn !
